src/DateTimePicker.jsx

```diff
@@ -187,7 +208,7 @@ class DateTimePicker extends React.Component {
 				<DateTimeLocalInput
 					id={id}
 					label={label}
-					value={this.state.datetime}
+					value={this.getDateTime()}
```

Comment:
this implementation forces an awkward separation of concerns - in effect, the parent of the `DateTimePicker` is responsible for both _setting_ and _reading_ its value. Form inputs should generally be [controlled components](https://facebook.github.io/react/docs/forms.html#controlled-components), where they are responsible for keeping track of their own value. The parent can then provide an `onChange` (or similar) function that receives the value each time it changes - at the moment it looks like essentially you are using the `onChangeCallback` function from the parent as _both_ a watcher and an indirect setter, which can make the app structure less modular and maintainable.